### PR TITLE
Add for multiple levels of Atx Headers in the CHANGELOG.md

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Should fix https://github.com/Azure/azure-sdk-for-java/issues/23908

### CODE REVIEW

File: eng/common/scripts/ChangeLog-Operations.ps1 [hunk 1]
```diff
@@ -42,14 +42,24 @@ function Get-ChangeLogEntriesFromContent {
   $changelogEntry = $null
   $sectionName = $null
   $changeLogEntries = [Ordered]@{}
+  $initialAtxHeader= "#"
```
Comment (weshaggard) [inline eng/common/scripts/ChangeLog-Operations.ps1:45]: What does `Atx` mean? Did you intend to put an `x` in there?

File: eng/common/scripts/Update-ChangeLog.ps1 [hunk 2]
```diff
@@ -113,7 +113,7 @@ if ($ReplaceLatestEntryTitle)
     {
         $entryToBeUpdated = Remove-EmptySections -ChangeLogEntry $entryToBeUpdated
     }
-    $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus -Content $entryToBeUpdated
+    $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus -InitialAtxHeader $ChangeLogEntries.InitialAtxHeader -Content $entryToBeUpdated
     LogDebug "Resetting latest entry title to [$($newChangeLogEntry.ReleaseTitle)]"
     $ChangeLogEntries.Remove($LatestVersion)
     if ($newChangeLogEntry) {
```
Comment (weshaggard) [inline eng/common/scripts/Update-ChangeLog.ps1:119]: See line 131 we have `##`.

File: eng/common/scripts/ChangeLog-Operations.ps1 [hunk 3]
```diff
@@ -261,11 +275,12 @@ function Set-ChangeLogContent {
     [Parameter(Mandatory = $true)]
     [String]$ChangeLogLocation,
     [Parameter(Mandatory = $true)]
-    $ChangeLogEntries
+    $ChangeLogEntries,
+    [String]$InitialAtxHeader="#"
```
Comment (weshaggard) [inline eng/common/scripts/ChangeLog-Operations.ps1:279]: Do we need to pass it in here? As we should have it on ChangeLogEntries already, if it isn't there we can default to #

File: eng/common/scripts/ChangeLog-Operations.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/common/scripts/ChangeLog-Operations.ps1:45]: Perhaps we call this `$baseHeaderIndention` or something like that.

Comment (chidozieononiwu) [reply eng/common/scripts/ChangeLog-Operations.ps1:45]: Thats the name of the Markdown headerstyle. At least acording to this doc https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md003---heading-style

**Review by weshaggard — APPROVED** [review]
The changes look reasonable. I suggest you run the different changelog operations on the java cosmos changelogs files to test.

**Pull request merged**